### ISSUE TITLE
Crazy amounts of new dependencies now that X-rays are merged

### ISSUE THREAD

**Issue by camsaul** [metabase/metabase#5711]
Running `lein uberjar` for the first time on this computer since X-rays were merged logged this:

```
Retrieving bigml/histogram/4.1.3/histogram-4.1.3.pom from clojars
Retrieving com/googlecode/json-simple/json-simple/1.1.1/json-simple-1.1.1.pom from central
Retrieving junit/junit/4.10/junit-4.10.pom from central
Retrieving org/hamcrest/hamcrest-core/1.1/hamcrest-core-1.1.pom from central
Retrieving org/hamcrest/hamcrest-parent/1.1/hamcrest-parent-1.1.pom from central
Retrieving kixi/stats/0.3.8/stats-0.3.8.pom from clojars
Retrieving redux/redux/0.1.3/redux-0.1.3.pom from clojars
Retrieving org/clojure/data.avl/0.0.17/data.avl-0.0.17.pom from central
Retrieving org/clojure/test.check/0.9.0/test.check-0.9.0.pom from central
Retrieving com/clearspring/analytics/stream/2.9.5/stream-2.9.5.pom from central
Retrieving com/addthis/common/build/maven/pom/jar-pom/3.4.0/jar-pom-3.4.0.pom from central
Retrieving com/addthis/common/build/maven/pom/papa-pom/2.1.0/papa-pom-2.1.0.pom from central
Retrieving it/unimi/dsi/fastutil/7.0.6/fastutil-7.0.6.pom from central
Retrieving org/slf4j/slf4j-api/1.7.7/slf4j-api-1.7.7.pom from central
Retrieving org/slf4j/slf4j-parent/1.7.7/slf4j-parent-1.7.7.pom from central
Retrieving redux/redux/0.1.4/redux-0.1.4.pom from clojars
Retrieving tide/tide/0.1.0-SNAPSHOT/tide-0.1.0-20160814.183356-11.pom from clojars
Retrieving org/clojure/clojure/1.9.0-alpha10/clojure-1.9.0-alpha10.pom from central
Retrieving com/github/brandtg/stl-java/0.1.1/stl-java-0.1.1.pom from central
Retrieving org/apache/commons/commons-math3/3.2/commons-math3-3.2.pom from central
Retrieving com/github/davidmoten/fastdtw/0.1/fastdtw-0.1.pom from central
Retrieving com/github/davidmoten/sonatype-parent/0.1/sonatype-parent-0.1.pom from central
Retrieving net/mikera/core.matrix/0.52.2/core.matrix-0.52.2.pom from clojars
Retrieving net/mikera/clojure-pom/0.6.0/clojure-pom-0.6.0.pom from clojars
Retrieving net/mikera/mikera-pom/0.6.0/mikera-pom-0.6.0.pom from central
Retrieving org/clojure/clojure/1.9.0-alpha5/clojure-1.9.0-alpha5.pom from central
Retrieving com/googlecode/json-simple/json-simple/1.1.1/json-simple-1.1.1.jar from central
Retrieving org/hamcrest/hamcrest-core/1.1/hamcrest-core-1.1.jar from central
Retrieving junit/junit/4.10/junit-4.10.jar from central
Retrieving org/clojure/data.avl/0.0.17/data.avl-0.0.17.jar from central
Retrieving com/clearspring/analytics/stream/2.9.5/stream-2.9.5.jar from central
Retrieving org/clojure/test.check/0.9.0/test.check-0.9.0.jar from central
Retrieving it/unimi/dsi/fastutil/7.0.6/fastutil-7.0.6.jar from central
Retrieving org/slf4j/slf4j-api/1.7.7/slf4j-api-1.7.7.jar from central
Retrieving com/github/brandtg/stl-java/0.1.1/stl-java-0.1.1.jar from central
Retrieving org/apache/commons/commons-math3/3.2/commons-math3-3.2.jar from central
Retrieving com/github/davidmoten/fastdtw/0.1/fastdtw-0.1.jar from central
Retrieving bigml/histogram/4.1.3/histogram-4.1.3.jar from clojars
Retrieving tide/tide/0.1.0-SNAPSHOT/tide-0.1.0-20160814.183356-11.jar from clojars
Retrieving redux/redux/0.1.4/redux-0.1.4.jar from clojars
Retrieving kixi/stats/0.3.8/stats-0.3.8.jar from clojars
Retrieving net/mikera/core.matrix/0.52.2/core.matrix-0.52.2.jar from clojars
```

We need to audit the dependencies of our dependencies and add appropriate `:exclusions` clauses. For example:

* We don't need (or want) `test.check` because we're using `expectations` for testing. And unit testing deps don't belong in the main uberjar anyway
* Similarly, we shouldn't need `junit`, or at the very least shouldn't be loading as a dep of the main uberjar

At any rate it looks like the biggest issue is some of our new deps are loading deps for their test code into our uberjar. This is bad because it makes the uberjar huge and launch slower (since it has to validate the JAR signature)

### Numbers

24.1 JAR has 54533 files (94MB)
master JAR has 71640 files (119MB) (edited)
so the new deps added 25 MB / 17,307 files (27% size increase)

**Comment by camsaul** [comment]
Also especially bad: Why are we loading Clojure 1.9 alpha? One of our deps may be overriding our core Clojure dependency

**Comment by camsaul** [comment]
We're actually loading two separate versions of Clojure 1.9 alpha. `lein deps :tree` tells me `1.8.0` is overriding those two. But fetching them is still a waste of bandwith.

Btw `lein deps :tree` makes recommendations about which exclusions you should add

**Comment by salsakran** [comment]
This is a blocker on 0.26 since it's breaking Java 7 compatibility, which I'd like to make for reasons other than dependency bloat. 

Also, we should try to be fairly svelte in our uberjar size, and push it down if possible. A 25mb jump is ... painful.

**Comment by camsaul** [comment]
Yeah that's a 27% size increase version-over-version

**Comment by sbelak** [comment]
Fixes in #5716 brought the size down to 100MB/58k files and we retain 1.7 compatibility.
